what4/src/What4/Interface.hs

```diff
@@ -1971,6 +1982,64 @@ class (IsExpr (SymExpr sym), HashableF (SymExpr sym)) => IsExprBuilder sym where
     pj <- realNe sym xi yi
     orPred sym pr pj
 
+-- | This newtype is necessary for @bvJoinVector@ and @bvSplitVector@.
+-- These both use functions from Data.Parameterized.Vector that
+-- that expect a wrapper of kind (Type -> Type), and we can't partially
+-- apply the type synonym (e.g. SymBv sym), whereas we can partially
+-- apply this newtype.
+newtype SymBV' sym w = MkSymBV' { getSymBV :: SymBV sym w }
+
+-- | Join a @Vector@ of smaller bitvectors
+bvJoinVector :: forall sym n w. (1 <= w, IsExprBuilder sym)
+             => sym
+             -> NatRepr w
+             -> Vector.Vector n (SymBV sym w)
+             -> IO (SymBV sym (n * w))
+bvJoinVector sym w = coerce $ Vector.joinWithM @IO @(SymBV' sym) @n bvConcat' w
+  where bvConcat' :: forall l. (1 <= l)
+                  => NatRepr l
+                  -> SymBV' sym w
+                  -> SymBV' sym l
+                  -> IO (SymBV' sym (w + l))
+        bvConcat' _ (MkSymBV' x) (MkSymBV' y) = MkSymBV' <$> bvConcat sym x y
+
+-- | Split a bitvector to a @Vector@ of smaller bitvectors
+bvSplitVector :: forall sym n w. (IsExprBuilder sym, 1 <= w, 1 <= n)
+              => sym
+              -> NatRepr n
+              -> NatRepr w
+              -> SymBV sym (n * w)
+              -> IO (Vector.Vector n (SymBV sym w))
+bvSplitVector sym n w x =
+  fmap (getSymBV @sym) <$>
```

Comment:
Here is where `coerce` would actually get you something

Reply:
Yep, changed!